### ISSUE TITLE
Locks Primary Flight Controls

### ISSUE THREAD

**Issue by ohhiitsmike** [till213/SkyDolly#107]
**Describe the bug**
I created a replay, played it back, and now my primary flight controls no longer work. All hardware is still functioning. Only time I get control is during or immediately after a replay. Going back to menu and starting a new flight causes the issue to occur again.

Have remapped controls, restarted the game/PC and even uninstalled SkyDolly. The problem is still there (facing a complete re-install).

**To Reproduce**
1. Create a replay
2. Play it back
3. Click "Stop" at the end or during
4. Control the plane as normal
5. Go back to menu and start new flight
6. Primary Flight Controls (Ailerons, Elevators, Rudder) not working from this point on.

**Expected behavior**
Create and play back replays and then be able to continue flying other flights without having to reinstall the entire game

**Screenshots**
N/A

**Application version:**
 - Oceania Update (Not Installed), Up to Date on Steam?

**Additional context**
Short of a complete reinstall and wiping out my Community Folder (I'm currently moving it elsewhere so I don't have any mods installed ("Safe Mode")), I have tried everything possible. Remaps, Reset to Defaults, Restarts etc.

**Comment by till213** [comment]
Hello,

Thank you for your detailed description, this is much clearer than what I understood from your initial description on flightsim.to.

So you are really talking about resuming control after a replay has ended (by clicking stop, or simply because it reached the end of the replay). As you say this works right after the replay("Only time I get control is during or immediately after a replay"), that is with the same aircraft ("in the same flight"), but not when you restart a complete new flight from within MSFS ("Going back to menu and starting a new flight causes the issue to occur again.").

What I am still not sure about, you wrote: "Have remapped controls, restarted the game/PC and even uninstalled SkyDolly. The problem is still there (facing a complete re-install)."

So first of all, "uninstalling" Sky Dolly (I assume you meant to have simply deleted its folder - Sky Dolly does not come yet with an installer that let's you uninstall the application, so some registry settings will still remain ;)) woudn't help at all. Because Sky Dolly does not install anything within your community folder, or anywhere else except in your Documents folder (the default location for the logbook) and Sky Dolly specific registry entries (search for "Sky Dolly" with regedit.exe if you wish). There is absolutely no modification of any MSFS related content / installation.

Second, and what I am not clear about: you say that the problem persists. So can you confirm that when you don't start a replay with Sky Dolly and you start a new flight, and then another one that your controls then work as you'd expect?

So:

A) WORKS:
* Start a new flight
* Go to main menu (the world map)
* Start another flight
*  Controls still work

B) DOES NOT WORK
* Start a new flight
* Sky Dolly: Start/stop a replay
* Go to main menu (the world map)
* Start another flight
*  Controls still NOT work

So can you confirm case A) above, that is the problem really only occurs when Sky Dolly did a replay?

Because if this is really only after a replay with Sky Dolly that would be really strange (especially when you say that the controls actually work right after a replay). I'd expect that MSFS would completely reset any flight data / controls, especially any SimConnect related data (Sky Dolly only communicates with MSFS via the official SimConnect API).

Also, could you answer the following:

* Which aircraft are you using (in case the problem is specific to one or several aircraft)?
* What kind of hardware are we talking about? Anything related to SimConnect perhaps (does your hardware require any additional software to be installed)?
* What about keyboard control? Doesn't that work either (if you'd map the keyboard to the yoke etc.)?
* Did you try other replay software like the free Flight Recorder (that essentially works in the exact same way like Sky Dolly, that is via SimConnect)?

I will try to reproduce this case (I do have a HOTAS throttle and joystick), but from your description this sounds very much like an issue with MSFS itself. And I am not even quite sure whether the Sky Dolly replay is involved here at all (but possibly due to its usage of SimConnect).

**Comment by ohhiitsmike** [comment]
Hi Oliver,
Apologies for any confusion

First of all, yes, all I did was delete the SkyDolly folder (which didn’t fix the issue). I have since reinstalled it again as it didn’t seem to change anything.

To put into better explanation, I was flying from Melilla to Almeria in the ATR72.
I had SkyDolly open from the moment the plane spawned but didn’t use it except when I was on short final for the destination so I could capture the landing.

I recorded and stopped recording without issue, and taxied to the gate and shut down the aircraft as normal. I then looked back at the replay and as soon as it stopped automatically, I quit the game (Main Menu, then Quit to Desktop).

I logged on the next day and I had primary flight control issues. No rudder, ailerons or elevators. I looked at the same replay from SkyDolly, only this time I manually stopped it, which allowed me to take control of the aircraft as I would normally expect. Thinking that was the problem solved and the game just “thought” it was still in replay mode.

I quit the flight and loaded up a new flight from the menu, again as normal, but I still had control issues. So I re-mapped my joystick and rudder just in case the bindings were messed up and found that game actually recognised my inputs, but was just not translating them to the aircraft (visible from the axis meters in the settings menu).

I have tried other aircraft, thinking it was just the ATR (since it launched as a buggy mess) and the PMDG 737, Fenix A320-214, FlyByWire A32NX and even base-game aircraft like the Asobo A20N and the Cessna 208 Caravan. All of them suffered the same issue. The only notable difference was the PMDG, which seemed to start fine, but then controls gradually slowed down. As though the game was turning down sensitivity by itself.

I then tried spawning at other areas and found something weird. By spawning on the runway itself, there is no problem whatsoever. However spawning anywhere else (parking ramp, gate, fuel box) re-creates the issue.

My last resort was to delete the community folder completely and run the game in safe mode. That didn’t rectify anything.

I can understand that as I go on, it sounds less and less like SkyDolly is the cause. However, I’m acting based on the fact that it has been the only install or change to my game for a good couple of months. And everything has been fine up until now.

It may not be an issue with the program itself, but maybe what it does to the game to function, which in turn broke in some way.

I’m hoping this is enough information for you to try and re-create the issue if that is your plan. I wish you the best of luck.

I really appreciate you help

Kind Regards,
Mike
